### ISSUE TITLE
Learning Rate incompatiblity

### ISSUE THREAD

**Issue by www3cam** [HIPS/autograd#253]
Hi I was running autograd on Keras with theano backend.  Currently Keras with the optimizer must do some sort of matrix assignment and I get this error (AutogradHint: This error *might* be caused by assigning into arrays, which autograd doesn't support.) when I try to optimize the learning rate for Keras.optimizers.Adam/Nadam/RMSprop.  I'm wondering if there is any way around this either via using your optimizer (in a keras framework) or anything else someone has found.  Thanks,

Cameron

**Comment by mattjj** [comment]
It's hard to say for sure without seeing the code you're running (or a simplified mock-up of it), but in general autograd doesn't work with Theano or Keras; it works with numpy, scipy, and any other numerical kernel library that one wraps in autograd primitives.
